nano/node/transport/message_deserializer.hpp

```diff
@@ -45,19 +42,20 @@ namespace transport
 
 		parse_status status;
 
-		message_deserializer (network_constants const &, network_filter &, block_uniquer &, vote_uniquer &);
+		using channel_read_fn_type = std::function<void (std::shared_ptr<std::vector<uint8_t>> const &, size_t, std::function<void (boost::system::error_code const &, std::size_t)>)>;
```

Comment:
The only thing I would change is name this better, it has nothing to do with channel, imo a short and suiting name is just `read_query_t`